lib/xandra.ex

```diff
@@ -218,133 +218,144 @@ defmodule Xandra do
+                           * idle_interval. Defaults to 30_000ms (30 seconds).
+                          """
+                        ],
+                        nodes: [
+                          type: {:list, :string},
+                          doc: """
+                            The Cassandra nodes to connect to. Each node in the list has to be in
+                            the form `"ADDRESS:PORT"` or in the form `"ADDRESS"`: if the latter is
+                            used, the default port (`#{@default_port}`) will be used for that node.
+                            Defaults to `["127.0.0.1"]`. This option must contain only one node.
+                            See the documentation for `Xandra.Cluster` for more information on
+                            connecting to multiple nodes.
+                          """
+                        ],
+                        pool_size: [
+                          type: :non_neg_integer,
+                          default: 1,
+                          doc: """
+                            The number of connections to start for the pool. Defaults to `1`, which
+                            means that a single connection is started.
+                          """
+                        ],
+                        port: [
+                          type: :non_neg_integer,
+                          doc: """
+                            Internal, not to be set by hand.
+                          """
+                        ],
+                        protocol_module: [
+                          type: {:custom, __MODULE__, :__validate_module__, ["protocol_module"]},
+                          doc: false
+                        ],
+                        protocol_version: [
+                          type: {:in, [:v3, :v4]},
+                          default: :v3,
+                          doc: """
+                            The version of the Cassandra native protocol to use, either `:v3` or
+                            `:v4`.
+                          """
+                        ],
+                        show_sensitive_data_on_connection_error: [
+                          type: :boolean,
+                          default: false,
+                          doc: """
+                            Is it ok to show sensitive data on connection errors? Useful for
+                            debugging and in tests.
+                          """
+                        ],
+                        transport_options: [
+                          type: :keyword_list,
+                          doc: """
+                            Options to forward to the socket transport. If `:encryption` is `true`,
+                            then the transport is SSL (see the Erlang `:ssl` module) otherwise it's
+                            TCP (see the `:gen_tcp` Erlang module).
+                          """
+                        ]
+                      )
+
```

Comment:
@whatyouhide `mix format` has made this pretty horrible. Any ideas about a setting to not do this?

Reply:
I ended up re-committing with the list defined as `@raw_schema` and then passing that as the arg to `NimbleOptions.new!/1`